syntax = "proto3";

    service TimeSeriesForecaster {
        rpc Forecast(ForecastRequest) returns (ForecastResponse);
    }

    message ForecastRequest {
        repeated DataPoint data = 1; // List of data points 
        int32 periods = 2;          // Number of periods to forecast
        string freq = 3;           // Frequency (e.g., "D" for daily)
        // Optional: Add Prophet parameters here later!
     }

     message DataPoint {
        string ds =  1;   // Date string (e.g., "2023-01-01")
        float y = 2;     // Value (e.g., 100.5)
     }

     message ForecastResponse {
        repeated ForecastPoint forecast = 1;  // List of forecasted points 
     }

     message ForecastPoint {
        string ds = 1;         // Date
        float yhat = 2;        // Predicted value
        float yhat_lower = 3;  // Lower bound
        float yhat_upper = 4;  // Upper bound
     }
